src/js/reducers/scalesReducer.ts

```diff
@@ -31,7 +31,13 @@ export function scalesReducer(state: ScaleState, action: ActionType<typeof scale
 
   if (action.type === getType(scaleActions.updateScaleProperty)) {
     const p = action.payload;
-    return state.setIn([String(id), p.property], p.value);
+    if (state.getIn([String(id), ...p.property.split(".")])) {
+      return state.setIn([String(id), ...p.property.split(".")], p.value);
+    }
+    const key = p.property.split(".").slice(-1).toString();
+    const obj = {};
+    obj[key] = p.value;
+    return state.setIn([String(id), ...p.property.split(".").slice(0,-1)], obj);
```

Comment:
these lines break things right now. to remind me, can you write down the cases that you're trying to handle here?

Reply:
Yeah this is what I needed some help with... Trying to handle the `range = {scheme: someColorScheme}` vs `range = someString`

Reply:
cool -- i think you can change this from an if statement to a try catch to handle the `TypeError`:
```
try {
      return state.setIn([String(id), ...p.property.split(".")], p.value);
    }
    catch (e) {
      const key = p.property.split(".").slice(-1).toString();
      const obj = {};
      obj[key] = p.value;
      return state.setIn([String(id), ...p.property.split(".").slice(0,-1)], obj);
    }
```

Reply:
and maybe leave a comment quickly explaining the typescript issue and reference the range and range.scheme example